### PR TITLE
use sdkmanager(1) instead of downloading compnents with wget(1)

### PR DESCRIPTION
**Pull request by gfx**
## Issue
- N/A

## Overview (Required)

- Android SDK tools provides [sdkmanager(1)](https://developer.android.com/studio/command-line/sdkmanager.html) that manages all the Android SDK components including `constraint-layout`
- `wget constraint-layout` sucks. Use sdkmanager instead.
- There are some circle-ci tweaks to make the build more stable

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -4,30 +4,26 @@ machine:
   environment:
     ANDROID_HOME: /usr/local/android-sdk-linux
     JAVA_OPTS: "-Xms512m -Xmx1024m"
-    GRADLE_OPTS: '-Dorg.gradle.parallel=false -Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
+    GRADLE_OPTS: '-Dorg.gradle.parallel=false -Dorg.gradle.daemon=false -Dorg.gradle.jvmargs="-Xmx1056m -XX:+HeapDumpOnOutOfMemoryError"'
```
Comment (gfx) [inline circle.yml]: Set `-Dorg.gradle.daemon=false`. Daemonize consumes too much memory.

File: circle.yml [hunk 2]
```diff
@@ -4,30 +4,26 @@ machine:
...
+    - $ANDROID_HOME/tools/bin/sdkmanager "platform-tools" "extras;android;m2repository" "extras;google;m2repository"
+    - $ANDROID_HOME/tools/bin/sdkmanager "extras;m2repository;com;android;support;constraint;constraint-layout;1.0.0-beta4"
+    - $ANDROID_HOME/tools/bin/sdkmanager "extras;m2repository;com;android;support;constraint;constraint-layout-solver;1.0.0-beta4"
   override:
-    - ./gradlew dependencies
+    - ./gradlew app:dependencies
 
 test:
   override:
-    - ./gradlew clean assembleProductionDebug check jacocoTestReportDevelop
```
Comment (gfx) [inline circle.yml]: Split tasks to reduce consuming JVM heap.

**Comment by konifar** [comment]
Thank you so much!! 👏

File: circle.yml [hunk 3]
```diff
@@ -4,30 +4,26 @@ machine:
...
-    - wget https://dl.google.com/android/repository/com.android.support.constraint-constraint-layout-1.0.0-beta4.zip -O /tmp/temp.zip && unzip /tmp/temp.zip -d /tmp/
-    - mv /tmp/constraintlayout $ANDROID_HOME/extras/android/m2repository/com/android/support/constraint/constraint-layout/1.0.0-beta4
-    - wget https://dl.google.com/android/repository/com.android.support.constraint-constraint-layout-solver-1.0.0-beta4.zip -O /tmp/temp.zip && unzip /tmp/temp.zip -d /tmp/
-    - mv /tmp/solver $ANDROID_HOME/extras/android/m2repository/com/android/support/constraint/constraint-layout-solver/1.0.0-beta4
-  cache_directories:
-    - ~/.android
-    - ~/.gradle
+    - $ANDROID_HOME/tools/bin/sdkmanager "platform-tools" "extras;android;m2repository" "extras;google;m2repository"
+    - $ANDROID_HOME/tools/bin/sdkmanager "extras;m2repository;com;android;support;constraint;constraint-layout;1.0.0-beta4"
+    - $ANDROID_HOME/tools/bin/sdkmanager "extras;m2repository;com;android;support;constraint;constraint-layout-solver;1.0.0-beta4"
```
Comment (konifar) [inline circle.yml]: 📝

**Pull request merged**